simbad/util/tests/test_pdb_util.py

```diff
@@ -78,6 +68,113 @@ def test_molecular_weight_3(self):
+        pdb_struct = PdbStructure.from_file(input_model)
+        data = pdb_struct.nres
+        reference_data = 59
+
+        self.assertEqual(data, reference_data)
+
+    def test_nres_2(self):
+        """Test case for PdbStructure.nres"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        data = pdb_struct.nres
+        reference_data = 192
+
+        self.assertEqual(data, reference_data)
+
+    def test_keep_first_chain(self):
+        """Test case for PdbStructure.keep_first_chain_only"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.keep_first_chain_only()
+        data = pdb_struct.nchains
+        reference_data = 1
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_idx_1(self):
+        """Test case for PdbStructure.select_chain_by_idx"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_idx(0)
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'A'
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_idx_2(self):
+        """Test case for PdbStructure.select_chain_by_idx"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_idx(1)
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'B'
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_id_1(self):
+        """Test case for PdbStructure.select_chain_by_id"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_id('A')
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'A'
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_id_2(self):
+        """Test case for PdbStructure.select_chain_by_id"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_id('B')
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'B'
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_residues(self):
+        """Test case for PdbStructure.select_residues"""
+        input_model = os.path.join(EXAMPLE_DIR, "toxd", "toxd.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        seqid_range = range(0, 5)
+        pdb_struct.select_residues(to_keep_idx=seqid_range)
+        data = pdb_struct.nres
+        reference_data = 5
+
+        self.assertEqual(data, reference_data)
```

Comment:
You only check that there are five residues, but not that the residues are in the desired range. You should check that `[residue.seqid for residue in pdb_struct.struct] == [1,2,3,4,5]`